clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -362,8 +362,14 @@ public MemoryRecords records() {
         return recordsBuilder.build();
     }
 
+    public int estimatedSizeInBytes() {
+        return recordsBuilder.estimatedSizeInBytes();
+    }
+
     public int sizeInBytes() {
-        return recordsBuilder.sizeInBytes();
+        if (!recordsBuilder.isClosed())
+            throw new IllegalArgumentException("Cannot get an exact size of the batch until it has been closed");
+        return records().sizeInBytes();
```

Comment:
Maybe we should remove this method? The one usage of this can just do `producerBatch.records().sizeInBytes()`?